control-plane/csi-driver/src/bin/node/format.rs

```diff
@@ -1,46 +1,42 @@
 //! Utility function for formatting a device with filesystem
-use csi_driver::context::FileSystem;
-
-use devinfo::blkid::probe::Probe;
-use std::process::Command;
-use tracing::{debug, trace};
-
-pub(crate) async fn prepare_device(device: &str, fstype: &FileSystem) -> Result<(), String> {
+use crate::filesystem_ops::{Ext4Fs, FileSystem, XFs};
+use csi_driver::filesystem::FileSystem as Fs;
+
+use tracing::debug;
+
+/// Prepare the filesystem before mount, change parameters if requested.
+pub(crate) async fn prepare_device(
+    fstype: &FileSystem,
+    device: &str,
+    staging_path: &str,
+    options: &[String],
+    fs_id: Option<&String>,
+) -> Result<(), String> {
     debug!("Probing device {}", device);
+    let fs = FileSystem::get_property(device, "TYPE");
 
-    let probe =
-        Probe::new_from_filename(device).map_err(|error| format!("probe setup failed: {error}"))?;
-
-    if let Err(error) = probe.do_probe() {
-        return Err(format!("probe failed: {error}"));
-    }
+    let fs_ops = fstype.fs_ops();
 
-    if let Ok(fs) = probe.lookup_value("TYPE") {
+    if let Ok(fs) = fs {
         debug!("Found existing filesystem ({}) on device {}", fs, device);
-        return Ok(());
-    }
+        if let Some(fs_id) = fs_id {
+            match fstype.0 {
```

Comment:
this should be part of the trait

Reply:
The parameters for both are different. That's why was not able to put in a trait. Now if we separate the repair from this, then repair again cannot be a part of trait.